### PR TITLE
Directly report logged errors instead of relying on rethrowing

### PR DESCRIPTION
**Pull request by cramforce**
…and then catching the `error` event.

Part of #7353

### CODE REVIEW

**Comment by erwinmombay** [comment]
having to do that workaround :(

File: src/log.js [hunk 1]
```diff
@@ -202,7 +215,7 @@ export class Log {
   error(tag, var_args) {
     const error = this.error_.apply(this, arguments);
     if (error) {
-      this.win.setTimeout(() => {throw /** @type {!Error} */ (error);});
+      reportError(error);
```
Comment (erwinmombay) [inline src/log.js]: does that mean we don't actually throw the error anymore?

Comment (cramforce) [inline src/log.js]: Yeah, do you think that is a problem?

Comment (cramforce) [inline src/log.js]: I'll merge for now. Considering to change our tests to always fail if an error was logged during execution (and requiring errors to be expected).

Comment (erwinmombay) [inline src/log.js]: sgtm. i don't think its a problem. just different expectation for dev that's all

**Pull request merged**